docs/source/guides/area_detectors.rst

```diff
@@ -0,0 +1,610 @@
+            """Simulate dark image collection for background subtraction."""
+            # Simulate dark collection process
+            original_num = self.cam.num_images.get()
+            self.cam.num_images.put(num_images)
+            self.cam.image_mode.put("Multiple")
+            # Implementation continues...
+
+.. note::
+   This pattern applies to any detector type. For production systems, replace
+   ``SimDetector`` with ``PilatusDetector``, ``PerkinElmerDetector``, etc.
+
+**Fast CCD Pattern:**
+
+.. code-block:: python
+
+    # devices/fastccd.py - Fast CCD configuration
+    from ophyd.areadetector import DetectorBase
+    from ophyd.areadetector.cam import FastCCDDetectorCam
+    from ophyd.areadetector.plugins import HDF5Plugin_V34
+    from ophyd import Component as Cpt
+
+    class FastCCDDetector(DetectorBase):
+        """Fast CCD detector with HDF5 file writing."""
+
+        cam = Cpt(FastCCDDetectorCam, ":cam1:")
+        hdf5 = Cpt(HDF5Plugin_V34, ":HDF1:",
+                   write_path_template="/data/%Y/%m/%d/")
+
+        def __init__(self, *args, **kwargs):
+            super().__init__(*args, **kwargs)
+
+            # Fast CCD specific configuration
+            self.cam.fccd_fw_enable.put(1)  # Enable firmware
+            self.cam.fccd_sw_enable.put(1)  # Enable software
+
+**Area Detector with Custom Processing:**
+
+.. code-block:: python
+
+    # devices/processing_detector.py - Custom image processing
+    from ophyd.areadetector import DetectorBase
+    from ophyd.areadetector.plugins import ProcessPlugin_V34, ROIPlugin_V34
+    from ophyd import Component as Cpt
+
+    class ProcessingDetector(DetectorBase):
+        """Detector with real-time image processing."""
+
+        # Multiple ROIs for different sample regions
+        roi1 = Cpt(ROIPlugin_V34, ":ROI1:")
+        roi2 = Cpt(ROIPlugin_V34, ":ROI2:")
+        roi3 = Cpt(ROIPlugin_V34, ":ROI3:")
+
+        # Image processing
+        proc1 = Cpt(ProcessPlugin_V34, ":Proc1:")
+
+        def setup_rois(self, sample_positions):
+            """Configure ROIs for different sample positions."""
+            for i, (roi, pos) in enumerate(zip([self.roi1, self.roi2, self.roi3],
+                                               sample_positions)):
+                roi.min_x.put(pos['x'] - pos['width']//2)
+                roi.min_y.put(pos['y'] - pos['height']//2)
+                roi.size_x.put(pos['width'])
+                roi.size_y.put(pos['height'])
+
+Plugin Configuration Patterns
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+**File Writing Plugins:**
+
+.. code-block:: python
+
+    # devices/file_writers.py - Advanced file writing
+    from ophyd.areadetector.plugins import HDF5Plugin_V34, TIFFPlugin_V34
+    from ophyd import Component as Cpt
+    from pathlib import Path
+    import datetime
+
+    class MultiFormatDetector(DetectorBase):
+        """Detector that saves in multiple formats."""
+
+        hdf5 = Cpt(HDF5Plugin_V34, ":HDF1:")
```

Comment:
There is a convention that the `:` in the PV name goes with the parent, not the child.  The prefix specifies `IOC:`, the child specifies `cam1:`, ...  Avoids the possibility and confusion that the delimiter is missing or repeated.  This convention starts with the EPICS IOC database files and continues through IOC startup, screen files, and other EPICS clients.  We should stick to this convention.

Reply:
```suggestion
        hdf5 = Cpt(HDF5Plugin_V34, "HDF1:")
```